<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GenerateDialog</class>
 <widget class="QDialog" name="GenerateDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>644</width>
    <height>422</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Generate Keyfile</string>
  </property>
  <property name="windowIcon">
   <iconset theme="QIcon::ThemeIcon::DocumentNew"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="font">
      <font>
       <pointsize>24</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Generate Keyfile</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="disclaimerGroupBox">
     <property name="title">
      <string>Disclaimer</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QLabel" name="disclaimerLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>This Generator uses the secrets module for generating cryptographically strong random numbers using the highest-quality sources provided by the operating System.

The quality of these random numbers may vary depending on your operating system and hardware, so you are free to use your own keys!</string>
        </property>
        <property name="textFormat">
         <enum>Qt::TextFormat::PlainText</enum>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="keyfileGroupBox">
     <property name="title">
      <string>Keyfile Settings</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QLabel" name="nameLabel">
        <property name="text">
         <string>Name:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLineEdit" name="keyNameEdit">
          <property name="placeholderText">
           <string>generated</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="keyEndingLabel">
          <property name="text">
           <string>.key</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="sizeLabel">
        <property name="text">
         <string>Size: (max: 4 GiB)</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="keyfileSizeLayout">
        <item>
         <widget class="QLineEdit" name="fileSizeEdit">
          <property name="inputMethodHints">
           <set>Qt::InputMethodHint::ImhDigitsOnly</set>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="placeholderText">
           <string>4</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="unitSelect">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="currentIndex">
           <number>3</number>
          </property>
          <item>
           <property name="text">
            <string>B</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>KiB</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MiB</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>GiB</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="generateButton">
          <property name="text">
           <string>Generate</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QProgressBar" name="generateProgressBar">
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="generateDialogSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <layout class="QHBoxLayout" name="buttonLayout_2">
     <item>
      <spacer name="buttonHorizontalSpacer">
       <property name="orientation">
        <enum>Qt::Orientation::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="doneGeneratingButton">
       <property name="text">
        <string>Done</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
